kit/ChildSession.cpp

```diff
@@ -3213,7 +3213,8 @@ void ChildSession::loKitCallback(const int type, const std::string& payload)
             else if (tokens.size() == 3 && tokens.equals(0, "EMPTY"))
             {
                 // with mode:    "EMPTY, <part>, <mode>"
-                const std::string part = (_docType != "text" ? tokens[1].c_str() : "0"); // Writer renders everything as part 0.
+                // Writer renders everything as part 0.
+                const std::string part = (_docType != "text" ? (_docType == "presentation" ? std::to_string(getLOKitDocument()->getPart()) : tokens[1].c_str()) : "0");
```

Comment:
hmm, this looks like we want to be smarter than core :)
I think the rule is "core is the only source of truth".
Why the core generates callback with wrong part number?
The proper fix shouldn't be where callback is emited?

Reply:
I checked the core side but it's a rabbit hole. This is a workaround until we have more time to check why the pViewShell pointer is null after a mode switch.

Reply:
That function still gets the information from core side. But there's a timing issue.

Reply:
I see..., one more thing: could you check if using this patch can we fix also master page switching?

Reply:
No, unfortunately it doesn't. I checked it.

Reply:
@gokaysatir this caused regression in slide previews updates I think, please see:
![invalidates](https://github.com/user-attachments/assets/7e34b1ba-d76f-4da5-a692-0ecfd96e9226)

NOW (top) vs OLD (bottom) on slide insertion